crowbar_framework/app/models/service_object.rb

```diff
@@ -730,7 +730,7 @@ def validate_at_least_n_for_role(proposal, role, n)
   def validate_count_as_odd_for_role(proposal, role)
     elements = proposal["deployment"][@bc_name]["elements"]
 
-    if not elements.has_key?(role) or elements[role].length.to_i.even?
+    if not elements.key?(role) or elements[role].length.to_i.even?
```

Comment:
Style/AndOr: Use `||` instead of `or`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)<br>Style/Not: Use `!` instead of `not`. (https://github.com/bbatsov/ruby-style-guide#bang-not-not)